Avito Product Analysis and Price Prediction

palak740 opened the pull request:
## Related Issue 
- Info about the related issue 
- [ ] LGM-SOC'21 Participant 
- [x] DevIncept Participant
- [x] Contributor

Closes: #475 

## Description of the project:
### Aim
Perform exploratory data analysis on the dataset of products from Avito Advertising website and develop a ML model to predict prices of the other products in the website

### Dataset used
https://www.kaggle.com/abderrahimalakouche/data-analysis-products-dataset

### Approach
* Cleaned the dataset, removed the duplicates and analysed it, worked on extracting various information through various plots.
* Encoded the categorial features and scaled the features before applying ML model to it.
* Applied various regression models and calculated new measure using R2 score, MAE, MSE.
* Predicted the price of test data using the best model.

### Models used:
* Random Forest Regressor
* Logistic Regression
* XGBoost Regressor
* Kernel Ridge
* Linear Regression
* Isolation Forest Regressor
* Support Vector Regressor
* CATBoost Regressor 
* LightGBM Regressor

### Libraries Needed:
* sklearn==0.0
* seaborn==0.10.1
* numpy==1.18.5
* pandas==1.0.5
* matplotlib==3.2.2
* scipy==1.5.0
* xgboost==1.4.2
* catboost==0.26
* lightgbm==3.2.1

### Conclusion:
* CATBoost Regressor's final score is found to be the highest among all the rest of the models. So, used the CATBoost Regressor for predicting price of the test dataset.

### Checklist:
- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x]  Any dependent changes have been merged and published in downstream modules.

prathimacode-hub reviewed (changes requested):
Your README.md is missing few things. Kindly check on to the README.md template provided.  @palak740

palak740 commented:
> Your README.md is missing few things. Kindly check on to the README.md template provided. @palak740

Dear mam, I have added a line and changed a heading, kindly check the same.
Rest, I guess, I have made the readme file based on the given template.

prathimacode-hub reviewed (approved):
One of the finest PR, it's exclusively presented. Great job. 🤩🙌@palak740

palak740 commented:
> One of the finest PR, it's exclusively presented. Great job. 🤩🙌@palak740

Thankyou mam:-)
